tests/test_size.py

```diff
@@ -1,170 +1,91 @@
+    box.size._width, box.size._height, box.size._ratio = BASE_SIZE
 
-        # Set the exact_width to the same value, but not as a minimum
-        self.size.width = 10
-        self.assertSize(self.size, (10, 2, 0.1))
+    assert_size(box.size, BASE_SIZE)
 
-        # Layout has been dirtied.
-        self.layout.dirty.assert_called_once_with(intrinsic_width=10)
+    return box
 
-        # Clean the layout
-        self.layout.dirty.reset_mock()
 
-        # Set the exact_width to something new
-        self.size.width = at_least(20)
-        self.assertSize(self.size, (at_least(20), 2, 0.1))
+def assert_size(size, values):
+    assert (size.width, size.height, size.ratio) == values
 
-        # Layout has been dirtied.
-        self.layout.dirty.assert_called_once_with(intrinsic_width=at_least(20))
 
-    def test_set_min_height(self):
-        self.size.height = at_least(10)
-        self.assertSize(self.size, (1, at_least(10), 0.1))
+def test_at_least_repr():
+    assert repr(at_least(10)) == "at least 10"
 
-        # Layout has been dirtied.
-        self.layout.dirty.assert_called_once_with(intrinsic_height=at_least(10))
 
-        # Clean the layout
-        self.layout.dirty.reset_mock()
+def test_size_repr(box):
+    assert repr(box.size) == "(1, 2)"
+    box.size.width = at_least(10)
+    assert repr(box.size) == "(at least 10, 2)"
 
-        # Set the exact height to the same value
-        self.size.height = at_least(10)
-        self.assertSize(self.size, (1, at_least(10), 0.1))
 
-        # Layout has NOT been dirtied.
-        self.layout.dirty.assert_not_called()
+@pytest.mark.parametrize("dimension", ["width", "height"])
+@pytest.mark.parametrize("minimum", [True, False])
+def test_set_dimension(box, dimension, minimum):
+    dim = at_least if minimum else lambda x: x
 
-        # Set the exact_height to the same value, but not as a minimum
-        self.size.height = 10
-        self.assertSize(self.size, (1, 10, 0.1))
+    setattr(box.size, dimension, dim(10))
+    assert_size(box.size, BASE_SIZE.change(dimension, dim(10)))
 
-        # Layout has been dirtied.
-        self.layout.dirty.assert_called_once_with(intrinsic_height=10)
+    # Layout has been dirtied.
+    box.layout.dirty.assert_called_once_with(**{f"intrinsic_{dimension}": dim(10)})
 
-        # Clean the layout
-        self.layout.dirty.reset_mock()
+    # Clean the layout
+    box.layout.dirty.reset_mock()
 
-        # Set the exact height to something else
-        self.size.height = at_least(20)
-        self.assertSize(self.size, (1, at_least(20), 0.1))
+    # Set the width to the same value
+    setattr(box.size, dimension, dim(10))
+    assert_size(box.size, BASE_SIZE.change(dimension, dim(10)))
 
-        # Layout has been dirtied.
-        self.layout.dirty.assert_called_once_with(intrinsic_height=at_least(20))
+    # Layout has NOT been dirtied.
+    box.layout.dirty.assert_not_called()
 
-    def test_set_ratio(self):
```

Comment:
Have we lost the ratio test in the port?

Reply:
Uhhh.... oops X(